### PR TITLE
fix(nodejs): address import issues in `lancedb` npm module

### PR DESCRIPTION
**Pull request by wjones127**
Fixes [#1496](https://github.com/lancedb/lancedb/issues/1496)

### CODE REVIEW

File: nodejs/lancedb/arrow.ts [hunk 1]
```diff
@@ -103,25 +103,12 @@ export type IntoVector =
...
-  | import("apache-arrow-15").Float
-  | import("apache-arrow-16").Float
-  | import("apache-arrow-17").Float;
-export type DataTypeLike =
-  | import("apache-arrow-13").DataType
-  | import("apache-arrow-14").DataType
-  | import("apache-arrow-15").DataType
-  | import("apache-arrow-16").DataType
-  | import("apache-arrow-17").DataType;
-
```
Comment (wjones127) [inline nodejs/lancedb/arrow.ts:118]: These cannot appear in our public API without forcing users to install all versions.

Comment (westonpace) [reply nodejs/lancedb/arrow.ts:118]: export function isDataType(value: unknown): value is DataType {
  return sanitizeType(value) instanceof DataType;
}

**Review by westonpace — COMMENTED** [review]
I don't think either approach works that well.

The former, as you pointed out, required all versions, and it still wouldn't work in all situations.
The latter, requires lock-step versioning, and that is extremely difficult to achieve in node.  

This is because, node will often install two copies of the same library, even if the versions are completely identical (rollup-style approaches basically inline all dependencies).

Can we use the `sanitizeType` method instead?

I added an example in a comment

File: nodejs/lancedb/arrow.ts (hunk 1 above)
Comment (westonpace) [reply nodejs/lancedb/arrow.ts:118]: (actually, I don't think we even need the `instanceof`, `sanitizeType` will throw if the input isn't type-like.

Comment (westonpace) [reply nodejs/lancedb/arrow.ts:118]: Or, (sorry for stream of consciousness), we should replace all calls to `isDataType` with `sanitizeType`.

Comment (wjones127) [reply nodejs/lancedb/arrow.ts:118]: No worries, I find this Typescript dependency resolution stuff very confusing so I appreciate the help.

**Review by westonpace — APPROVED** [review]
Tests pass and this matches my knowledge to the extent I understand anything.  I assume whatever integration test you are testing with is also working?

**Comment by wjones127** [comment]
> Tests pass and this matches my knowledge to the extent I understand anything. I assume whatever integration test you are testing with is also working?

Indeed. Just re-ran it with these changes. I can install a local version of this.

<details><summary>Details</summary>
<p>

Create a local version

```shell
npm run build:release
npm run build-release
npm pack
```

Then you can use that package elsewhere. It runs successfully with type checking. However, I haven't figured out how to pack the native module yet, so it doesn't work at runtime 🤷 

</p>
</details>

**Pull request merged**